.github/workflows/prepare_release.yml

```diff
@@ -0,0 +1,72 @@
+---
+
+name: Prepare Release
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: 'Module version to be released.'
+        required: false
+        default: ''
+        type: string
+      working-directory:
+        description: The working directory where all jobs should be executed. Used for modules in subdirectories like a monorepo or a control repository.
+        default: '.'
+        required: false
+        type: string
+      base-branch:
+        description: 'The branch that will be used as the origin for the release branch.'
+        required: false
+        default: ''
+        type: string
+
+env:
+  BUNDLE_WITHOUT: development:test:system_tests
+  BUNDLE_WITH: release
+  GIT_AUTHOR_NAME: Release Automation
+  GIT_AUTHOR_EMAIL: "${{ github.repository_owner }}@users.noreply.github.com"
+  BLACKSMITH_FULL_VERSION: "${{ inputs.version }}"
+
+jobs:
+  prepare_release:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    name: 'Prepare Release'
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.base-branch }}
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3'
+          bundler-cache: true
+          working-directory: ${{ inputs.working-directory }}
+      - name: bundle environment
+        run: |
+          bundle env
+          bundle list
+      - name: Update metadata.json to new version
+        run: |
+          if [[ -n BLACKSMITH_FULL_VERSION ]] ; then
+            bundle exec rake module:bump:full
+          else
+            bundle exec rake module:bump
+          fi
+      - name: Prepare the release
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          CHANGELOG_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: bundle exec rake release:prepare
+      - name: Create pull Request
```

Comment:
Looking at https://github.com/voxpupuli/puppet-example/pull/82 I wonder if we can come up with some better instructions in the body. For example, it could link to https://voxpupuli.org/docs/releasing_version/ though that isn't really explaining it well for this new process.

Reply:
My idea was to link to  https://voxpupuli.org/docs/releasing_version/ and update it